### ISSUE TITLE
Support the different standard encodings

### ISSUE THREAD

**Issue by michaelWuensch** [alexzhirkevich/custom-qr-generator#37]
**Is your feature request related to a problem? Please describe.**
I want to be my QR codes as small as possible.
I have to encode 64 character hex strings.

**Describe the solution you'd like**
The QR code specification defines 4 standard ways of encoding the data.
Numeric, Alphanumeric, binary & kanji.
I could encode hex strings way more efficiently if I just upper case them and then use Alphanumeric as encoding.
So in short:
It would be awesome if we could set the encoding option somehow.

**Describe alternatives you've considered**
The alterative are ugly big codes 😀

**Additional context**
Please see this page for more information:
https://www.thonky.com/qr-code-tutorial/data-encoding

**Comment by alexzhirkevich** [comment]
I guess ZXing encoder [automatically picks](https://github.com/zxing/zxing/blob/95c9987cca22d4c7812a774531011edd8ec8a4f7/core/src/main/java/com/google/zxing/qrcode/encoder/Encoder.java#L111) most suitable mode.

**Comment by michaelWuensch** [comment]
Hm, okay. I tried it with a 64 char hex string that I uppercased and at least it did not become smaller (same amount of pixels). I haven't checked though if the encoded data was actually different. I will give it another go later.

**Comment by alexzhirkevich** [comment]
Large codes is a bug with automatic error correction level. It mistakenly decides that you have logo with default size even when you don't have it and increases ECL. Try change `errorCorrectionLevel` in options from `Auto` to `Low`. I think it will help to reduce the QR code size.

**Comment by michaelWuensch** [comment]
I do not use auto for error correction. I use `Medium` for for that 64 char string. Setting it to `low` made a difference in size of the QR code while uppercasing the hex string had no effect.
Right now I need to stick with medium error correction to have them scan reliably.
I'll do some more test later.

**Comment by alexzhirkevich** [comment]
So [this method](https://github.com/zxing/zxing/blob/95c9987cca22d4c7812a774531011edd8ec8a4f7/core/src/main/java/com/google/zxing/qrcode/encoder/Encoder.java#L246C5-L270) form ZXing should automatically select alphanumeric mode. It can't be controlled by the user.

**Comment by michaelWuensch** [comment]
Yes, I have already seen that function and it looks good.
I will close the issue for now and only reopen if I can proof it does actually not use the correct encoding.
Thanks for your support! 👍

**Comment by michaelWuensch** [comment]
@alexzhirkevich 
I just looked at it again and I can confirm that it works and uses the standard encodings automatically.
I added the .toUppercase() function on the wrong line and it was actually still lower case...
Big facepalm moment! Sorry for wasting your time.
But the good news is everything works as expected!
